vendor/github.com/OpenBazaar/wallet-interface/wallet.go

```diff
@@ -228,6 +228,12 @@ type Wallet interface {
 	// owned by this wallet, or being watched by the wallet via AddWatchedAddress method.
 	AddTransactionListener(func(TransactionCallback))
 
+	// CallTransactionListener is how openbazaar-go invokes callbacks whenever
+	// a transaction is received that is relevant to this wallet or any of its watch only
+	// addresses. An address is considered relevant if any inputs or outputs match an address
+	// owned by this wallet, or being watched by the wallet via AddWatchedAddress method.
+	CallTransactionListeners(callback TransactionCallback)
```

Comment:
This doc is misleading. Openbazaar-go does not normally invoke the wallet when a transaction is detected as most wallets don't require this behavior. Usually the wallet can detect its own transactions through AddWatchedAddress. Really, what this wants to be is some interface for updating the state of specific entries in the utxo ledger that ETH shares with other utxo wallets. The fact that non-ETH wallets aren't implementing the `CallTransactionListeners` method is another signal that we're overloading some internal behavior here.

I'm not sure what the fix looks like here, but I'd like to discuss how we're approaching the ledger for non-utxo wallets and how we can make the interface (and this method specifically) better purposed for these wallets without surprising behavior from wallets which don't _need_ certain methods. (More specifically: I don't want to call `CallTransactionListeners` and find that they didn't do anything for BTC because of internal implementation details "didn't require it".)

Reply:
@cpacia @placer14 
Currently we have `AddTransactionListeners` which basically sets/adds a function to process a `TransactionCallback` struct. 
The existing UTXO wallets use : https://github.com/OpenBazaar/openbazaar-go/blob/f50b53ef99e7bc87707eda6ed0f3a8b114d28588/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go#L506 and https://github.com/OpenBazaar/openbazaar-go/blob/f50b53ef99e7bc87707eda6ed0f3a8b114d28588/vendor/github.com/OpenBazaar/multiwallet/service/wallet_service.go#L511 to process incoming transactions. Since this behavior is not available for non-utxo cryptos, there has to be a more general way to invoke the callbacks.
`CallTransactionListeners` is a get method for the added or set transaction listeners.
Having this would decouple the responsibility of server to update the state independent of the wallet's internal processing. A wallet implementation should be responsible for the construction of the `TransactionCallback` struct and then hand over the processing to the server. This is what I am trying to accomplish here.